core/chaincode/shim/crypto/ac/abac.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ac
+
+import (
+	"bytes"
+	"crypto/x509"
+	"errors"
+
+	"github.com/hyperledger/fabric/core/crypto/abac"
```

Comment:
Agree with @srderson, @muralisrini and @diegomasini. I understand the problem but in my opinion duplicate code is not a good option. I think that a new issue could be opened to treat this problem in a more general way. I think that a possible solution could be structure a few core libraries to be imported into the shim.

For example attributes have to be processed in three places client, membership services and here. To do that I have defined the abac library, this library is imported from the three places, and defines the core functionality required by abac as for example derivates an attribute key. Probably we could redefine this library to reduce the size and move it to other place, but in my opinion a core library used from the three places is required. Currently is located in the same place that other libraries like "primitives" or "utils" in similar situation.